include/ebpf_api.h

```diff
@@ -417,6 +417,35 @@ extern "C"
     _Must_inspect_result_ ebpf_result_t
     ebpf_get_program_info_from_verifier(_Outptr_ const ebpf_program_info_t** program_info);
 
+    /**
+     * @brief Run the program in the eBPF VM, measure the execution time, and return the result.
+     *
+     * @param[in] program_fd File descriptor of the program to run.
+     * @param[in] input_size Size of the input data.
+     * @param[in] input Pointer to the input data.
```

Comment:
As noted in the references in issue #715 there are two inputs (ctx_in and data_in), and two outputs (ctx_out and data_out):

https://www.kernel.org/doc/html/latest/userspace-api/ebpf/syscall.html
> Run the eBPF program associated with the prog_fd a repeat number of times against a provided program context ctx_in and data data_in, and return the modified program context ctx_out, data_out (for example, packet data), result of the execution retval, and duration of the test run.
>
> The sizes of the buffers provided as input and output parameters ctx_in, ctx_out, data_in, and data_out must be provided in the corresponding variables ctx_size_in, ctx_size_out, data_size_in, and/or data_size_out. If any of these parameters are not provided (ie set to NULL), the corresponding size field must be zero.

https://www.mankier.com/8/bpftool-prog
> bpftool prog run PROG data_in FILE [data_out FILE [data_size_out L]] [ctx_in FILE [ctx_out FILE [ctx_size_out M]]] [repeat N]

I don't see how you encode both of them without separate arguments for each.  It looks to me like this PR only supports data_in and data_out and not ctx_in and ctx_out.

Reply:
Added support for passing both data and context.